bin/main.ml

```diff
@@ -0,0 +1,194 @@
+open Eio.Std
+
+let or_die = function
+  | Ok x -> x
+  | Error `Msg m -> failwith m
+
+let pp_timestamp f x =
+  let open Unix in
+  let tm = localtime x in
+  Fmt.pf f "%04d-%02d-%02d %02d:%02d.%02d" (tm.tm_year + 1900) (tm.tm_mon + 1)
+    tm.tm_mday tm.tm_hour tm.tm_min tm.tm_sec
+
+let reporter =
+  let report src level ~over k msgf =
+    let k _ =
+      over ();
+      k ()
+    in
+    let src = Logs.Src.name src in
+    msgf @@ fun ?header ?tags:_ fmt ->
+    Fmt.kpf k Fmt.stderr
+      ("%a %a %a @[" ^^ fmt ^^ "@]@.")
+      pp_timestamp (Unix.gettimeofday ())
+      Fmt.(styled `Magenta string)
+      (Printf.sprintf "%14s" src)
+      Logs_fmt.pp_header (level, header)
+  in
+  { Logs.report }
+
+let setup_log style_renderer level =
+  Fmt_tty.setup_std_outputs ?style_renderer ();
+  Logs.set_level level;
+  (* Disable tls.tracing when logs are set to debug *)
+  (* List.iter
+     (fun src -> match Logs.Src.name src with "tls.tracing" -> Logs.Src.set_level src (Some Info) | _ -> ())
+     @@ Logs.Src.list (); *)
+  Logs.set_reporter reporter;
+  ()
+
+let export service ~on:socket =
+  Lwt_eio.run_lwt @@ fun () ->
```

Comment:
If you use your Eio port of Capnp, we wouldn't have any Lwt dependency at all to this solver...

Reply:
I'd like to go fully Eio if we can.

Reply:
We're also using it for git_unix.